### ISSUE TITLE
Save struct (perhaps through Serde?)

### ISSUE THREAD

**Issue by cazgp** [netvl/ejdb.rs#2]
It would be really nice if one could do something like

```
struct Model {
    title: String,
    body: String,
}

let m = Model { title: "Title", body: "Body" };
coll.save(bson!{m});
# or even
coll.save(&m);
```

One way I attempted to do this was by using Serde:

```
use bson::to_bson;

#[derive(Serialize, Deserialize, Debug)]
struct Model {
    title: String,
    body: String,
}
let m = Model { title: "Title", body: "Body" };
coll.save(to_bson(&m).unwrap());
```

... and a variety of other things including `from_bson` and `EjdbDocument`.

I'm a Rust noob, so attempting to integrate this with Serde is beyond my understanding for now. Do you have any other suggestions as to how this could work?

**Comment by netvl** [comment]
This is something which should be provided by the [bson-rs](https://github.com/zonyitoo/bson-rs) library. It appears it has some kind of support for automatic serialization through serde and rustc-serialize. Check whether it supports your use case; if it does, please write here and I'll bump the bson-rs dependency version if necessary.

**Comment by netvl** [comment]
Ah, I see. It indeed does have the serialization support in 0.3, but ejdb-rs depends on 0.2. I'll bump the version number and release the new version when I'll be able to, probably today or tomorrow.

**Comment by cazgp** [comment]
The second example I gave is me trying to get `bson-rs` to work. That attempt fails with the error: 

```
53 |         coll.save(to_bson(&self).unwrap());
   |              ^^^^ trait `bson::Bson: std::borrow::Borrow<ejdb::bson::ordered::OrderedDocument>` not satisfied
```

**Comment by netvl** [comment]
Yep, it happens because ejdb-rs depends on bson-0.2 but you seem use `to_bson()` from `bson` crate directly, instead of `ejdb::bson` module (which reexports the `bson` crate the library depends on).

**Comment by cazgp** [comment]
Gotcha. OK. Looking forward to a new version number, and thank you :)

**Comment by netvl** [comment]
Fixed by #3.
